lms/djangoapps/commerce/views.py

```diff
@@ -65,6 +67,12 @@ def checkout_receipt(request):
             "If your course does not appear on your dashboard, contact {payment_support_link}."
         ).format(payment_support_link=payment_support_link)
 
+    commerce_configuration = CommerceConfiguration.current()
+    # user order cache should be cleared when a new order is placed so user can see new order
+    if is_payment_complete and commerce_configuration.enabled and commerce_configuration.is_cache_enabled:
+        cache_key = commerce_configuration.CACHE_KEY + '.' + request.user.username
```

Comment:
Please use the user ID instead of the username to cache this information. The username is public and goes through a hash function [here](https://github.com/edx/edx-platform/blob/master/common/djangoapps/util/memcache.py#L27). If someone's username is too long and needs to be hashed, it would be possible to engineer a username which would allow you to take another user's order history from the cache. This can be avoided by using information that's not public in the cache key, like the user's ID.